### ISSUE TITLE
Adding/removing backdrop while sidebar open

### ISSUE THREAD

**Issue by shiptoncraig** [arkon/ng-sidebar#45]
I am using ng-sidebar in a project and need to be able to add/remove the backdrop while a sidebar is already open.  Is this something you've ever considered implementing?

**Comment by arkon** [comment]
That could be added. Would you be toggling the `showBackdrop` input on the sidebar, or would it make more sense to have a global flag on the container component?

**Comment by shiptoncraig** [comment]
Yeah, my first instinct before I looked at the source was to try using [showBackdrop] on ng-sidebar to change it programmatically.  

That being said, as the ng-sidebar-container component already contros the backdrop and its styles, it seems like a good place to put it.

**Comment by shiptoncraig** [comment]
Have you been able to use npm link when developing locally?  I keep finding that angular can't import the SidebarModule when ;linking.

**Comment by arkon** [comment]
I was actually using a pretty convoluted way of reinstalling the package, but I think linking it should work too. Did you build the main project first?

**Comment by shiptoncraig** [comment]
It builds just before linking, so I thought that would do it.  There's nothing particularly different at first glance between 3.2.2 hosted on the public NPM registry and the locally built one.  

Does an `npm publish` do everything it needs to when you push to the NPM registry?

**Comment by arkon** [comment]
Yes, it builds the project first.

**Comment by archs147** [comment]
Hi

I would like to know when this enhancement was added. I am working with version 6.0,4 and I am trying to show the backdrop when the sidebar is open but it doesn't work.

please your help.

regards

**Comment by arkon** [comment]
@archs147 It was added quite a while back. What does your template look like?

**Comment by archs147** [comment]
<ng-sidebar
        [(opened)]="_viewSideBar"
        [position]="'right'"
        [(showBackdrop)]="_showBackdrop"
        [mode]="'over'">

I set the _showBackdrop=true when the side bar is open but it doesn't work, then when I close the sidebar and open it again the backdrop appears.

regards

**Comment by archs147** [comment]
Hi @arkon 

Any idea about where is the problem?

thanks

**Comment by arkon** [comment]
I can't reproduce this. How are you setting the `open` and `showBackdrop` states?
